master/buildbot/test/unit/test_process_properties.py

```diff
@@ -1232,6 +1232,42 @@ def testFlattenListAdd2(self):
                       ["string", "bla", "string", "bla"])
         return d
 
+    @defer.inlineCallbacks
+    def testCompEq(self):
+        self.props.setProperty("do-tests", "string", "scheduler")
+        result = yield self.build.render(Property("do-tests") == "string")
+        self.assertEqual(result, True)
```

Comment:
need a test for
```python
        result = yield self.build.render("string" == Property("do-tests"))
```
Not sure if it will work, but at least it should be tested (maybe put it unittest.expectedFailure)